examples/pytorch/graphsaint/train_sampling.py

```diff
@@ -0,0 +1,204 @@
+
+        # evaluate
+        if epoch % args.val_every == 0:
+            val_f1_mic, val_f1_mac = evaluate(
+                model, g, labels, val_mask, multilabel)
+            print(
+                "Val F1-mic {:.4f}, Val F1-mac {:.4f}".format(val_f1_mic, val_f1_mac))
+            if val_f1_mic > best_f1:
+                best_f1 = val_f1_mic
+                print('new best val f1:', best_f1)
+                torch.save(model.state_dict(), os.path.join(
+                    log_dir, 'best_model.pkl'))
+
+    end_time = time.time()
+    print(f'training using time {end_time - start_time}')
+
+    # test
+    if True:
+        model.load_state_dict(torch.load(os.path.join(
+            log_dir, 'best_model.pkl')))
+    test_f1_mic, test_f1_mac = evaluate(
+        model, g, labels, test_mask, multilabel)
+    print("Test F1-mic {:.4f}, Test F1-mac {:.4f}".format(test_f1_mic, test_f1_mac))
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='GCN')
+    # data source params
+    parser.add_argument("--dataset", type=str, choices=['ppi', 'flickr', 'reddit', 'yelp', 'amazon'], default='ppi',
+                        help="Name of dataset.")
+
+    # cuda params
+    parser.add_argument("--gpu", type=int, default=-1,
+                        help="GPU index. Default: -1, using CPU.")
+
+    # sampler params
+    parser.add_argument("--sampler", type=str, default="node", choices=['node', 'edge', 'rw'],
+                        help="Type of sampler")
+    parser.add_argument("--node-budget", type=int, default=6000,
+                        help="Expected number of sampled nodes when using node sampler")
+    parser.add_argument("--edge-budget", type=int, default=4000,
+                        help="Expected number of sampled edges when using edge sampler")
+    parser.add_argument("--num-roots", type=int, default=3000,
+                        help="Expected number of sampled root nodes when using random walk sampler")
+    parser.add_argument("--length", type=int, default=2,
+                        help="The length of random walk when using random walk sampler")
+    parser.add_argument("--num-repeat", type=int, default=50,
+                        help="Number of repeating sampling one node to estimate edge / node probability")
+
+    # model params
+    parser.add_argument("--n-hidden", type=int, default=512,
+                        help="Number of hidden gcn units")
+    parser.add_argument("--arch", type=str, default="1-0-1-0",
+                        help="Network architecture. 1 means an order-1 layer (self feature plus 1-hop neighbor "
+                             "feature), and 0 means an order-0 layer (self feature only)")
+    parser.add_argument("--dropout", type=float, default=0,
+                        help="Dropout rate")
+    parser.add_argument("--batch-norm", action='store_true',
+                        help="Whether to use batch norm")
```

Comment:
From README, it seems that `batch_norm` is always True. If so, maybe you can change this to `no-batch-norm` and update the README file.